config/configuration_service.proto

```diff
@@ -7,65 +7,128 @@ package config;
     //Used when you just need to read the current configuration
-    rpc GetConfiguration(ReadRequest) returns (ConfigurationResponse) {}
+    rpc GetConfiguration (ReadRequest) returns (ConfigurationResponse) {
+    }
 
     //Used when you want to update the existing configuration
-    rpc UpdateConfiguration(UpdateRequest) returns (ConfigurationResponse) {}
+    rpc UpdateConfiguration (UpdateRequest) returns (ConfigurationResponse) {
+    }
 
     //This is used to instruct the Daemon to stop taking in any new requests,
     //At this point , however any existing requests being processed will be honored.
     // ("_Request_Stop", "block_number",authentication_address) should be sent in the signature
-    rpc StopProcessingRequests(CommandRequest) returns (Response) {}
+    rpc StopProcessingRequests (CommandRequest) returns (Response) {
+    }
 
     //Restores the normal behavior to start processing new requests again
     // ("_Request_Start", "block_number",authentication_address) should be sent in the signature
-    rpc StartProcessingRequests(CommandRequest) returns (Response) {}
-}
+    rpc StartProcessingRequests (CommandRequest) returns (Response) {
+    }
+
+    //To Determine if the Daemon is processing requests are not.
+    // ("__Is_Daemon_Processing_Requests", "block_number",authentication_address) should be sent in the signature
+    rpc IsDaemonProcessingRequests (ReadRequest) returns (Response) {
+    }
+
 
 
+}
+
 //Used when you just need to read the current configuration
 message ReadRequest {
     //Signature will compromise of the below
     // ("_Request_Read", "block_number",authentication_address)
+    //OR
+    //("__Is_Daemon_Processing_Requests", "block_number",authentication_address)
     bytes signature = 1;
 
+    //current block number (signature will be valid only for short time around this block number)
+    uint64 current_block = 2;
+
+    //The address of the person who has rights to make configuration changes on Daemon
+    string user_address = 4;
+
 }
 
+
+
 //Used when you want to update the existing configuration
 message UpdateRequest {
     //Signature will compromise of the below
-    // ("_Request_Update", "block_number","updated_json_data",authentication_address)
+    // ("_Request_Update", "block_number",authentication_address)
     bytes signature = 1;
-    //Holds the updated configuration in the form of JSON,only the fields to be updated should be sent
-    string updated_json_data = 2;
+    //Indicates the updated configuration ( only the modified leaf and its changed value is passed ) Example (log.output.rotation_time_in_sec is a valid key )
+    map<string, string> updated_configuration = 2;
 
-}
+    //current block number (signature will be valid only for short time around this block number)
+    uint64 current_block = 3;
+
+    //The address of the person who has rights to make configuration changes on Daemon
+    string user_address = 4;
 
+}
 
 message CommandRequest {
```

Comment:
Both `ReadRequest` and `CommandRequest` contain only authentication information. If there is no reason to keep them separate we could replace them by single request:
```proto
//Used with methods which don't have other parameters except authentication data
message EmptyRequest {
    //Caller authentication data
    CallerAuth auth = 1;
}
```